src/aks-preview/azext_aks_preview/_validators.py

```diff
@@ -213,6 +213,20 @@ def validate_eviction_policy(namespace):
             raise CLIError("--eviction-policy can only be Delete or Deallocate")
 
 
+def validate_spot_max_price(namespace):
+    """Validates the spot node pool max price."""
+    if not isnan(namespace.spot_max_price):
+        if namespace.priority != "Spot":
+            raise CLIError("--spot_max_price can only be set when --priority is Spot")
+        if namespace.priority == "Spot":
+            if namespace.spot_max_price * 100000 % 1 != 0:
+                raise CLIError("--spot_max_price can only include up to 5 decimal places")
```

Comment:
I suggest add these rules to help. We can let users know in advance.

Reply:
added

Reply:
added.